flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImplTest.java

```diff
@@ -776,7 +686,9 @@ public void testOrphanedAllocationIsCanceledIfNotRemapped() throws Exception {
 		final List<AllocationID> canceledAllocations = new ArrayList<>();
 		resourceManagerGateway.setCancelSlotConsumer(canceledAllocations::add);
 
-		try (SlotPoolImpl slotPool = createAndSetUpSlotPool()) {
+		try (SlotPoolImpl slotPool = createAndSetUpSlotPool(resourceManagerGateway)) {
+			slotPool.registerTaskManager(taskManagerLocation.getResourceID());
```

Comment:
`registerTaskManager` is not need because the following `registerAndOfferSlot` will do it.